### DISCUSSION TITLE
Q: pikaraoke functionality: support for mp3 vocal/instrument separation and lyric generation?

### DISCUSSION THREAD

**Discussion by sensboston** [vicwomg/pikaraoke#316] [General]
I'm working on a karaoke system for Windows (I have a bunch of different RPs at home but my main "media player" (connected to a large living room TV) is a Windows PC (small, fanless but powerful, also used as a home server with a bunch of services etc.), so I am interested in any implementation of a karaoke system, multi-platform or only for Windows.

So my question is: is pikaraoke just a karaoke player and requires specially prepared songs (or just plays karaoke style videos from YouTube) or can it play karaoke from any mp3 file (i.e. perform vocal and instrument separation, extract text by time, etc.) ?

**Comment by vicwomg** [comment]
Pikaraoke does not do mp3 vocal/instrument separation. It supports youtube, video files, and mp3+cdg

**Reply by sensboston** [reply]
Thank you for the confirmation
P.S. By the way, where are you grab the timed song lyrics?  Currently I'm using Whisper https://github.com/openai/whisper but it's slow and sometimes results aren't pretty correct.

**Reply by vicwomg** [reply]
Pikaraoke does not grab timed song lyrics. They are encoded into the video or the cdg files. These are created by karaoke content authors. 

Think of pikaraoke as a front end interface for karaoke content from youtube, and your own karaoke library files. It does not generate karaoke files nor content. It's a "KTV" style system for multiple users to queue songs on a TV.

**Reply by vicwomg** [reply]
That's not to say it couldn't be a future addition, but I don't think there's a good library for this yet (much less one that would perform well on a raspberry pi)

**Reply by sensboston** [reply]
Yeah, agree: both [vocal-remover](https://github.com/tsurumeso/vocal-remover/) and [Whisper ](https://github.com/openai/whisper) are "heavy" even for regular Windows PC with not soo old CPU and definitely will not work satisfyingly on RPis (but might be in future...)

BTW, if you're interested, I can ship you an executable or provide an access to my private repo "UltimateKaraoke" 🤣 (still not decided to open it) when I reach an early beta stage.
